pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/DestinationLookup.java

```diff
@@ -100,16 +99,17 @@ public void lookupDestinationAsync(@PathParam("property") String property, @Path
             return;
         }
 
-        CompletableFuture<LookupResult> lookupFuture = pulsar().getNamespaceService().getBrokerServiceUrlAsync(topic,
+        CompletableFuture<Optional<LookupResult>> lookupFuture = pulsar().getNamespaceService().getBrokerServiceUrlAsync(topic,
                 authoritative);
 
-        lookupFuture.thenAccept(result -> {
-            if (result == null) {
+        lookupFuture.thenAccept(optionalResult -> {
+            if (optionalResult == null || !optionalResult.isPresent()) {
```

Comment:
minor comment.. if `optionalResult` is `Optional` then should we just check `!optionalResult.isPresent()` and make sure that which-ever method returns this result always return non-null result.